$MirOS: ports/emulators/qemu/stable/patches/patch-osdep_h,v 1.1 2008/05/07 15:46:11 tg Exp $
--- osdep.h.orig	Wed Apr 27 20:51:43 2005
+++ osdep.h	Tue May  3 18:04:45 2005
@@ -2,6 +2,10 @@
 #define QEMU_OSDEP_H
 
 #include <stdarg.h>
+#ifdef __OpenBSD__
+#include <sys/types.h>
+#include <sys/signal.h>
+#endif
 
 int qemu_vsnprintf(char *buf, int buflen, const char *fmt, va_list args);
 void qemu_vprintf(const char *fmt, va_list ap);
@@ -31,7 +35,7 @@ extern void __longjmp(jmp_buf env, int v
 struct qemu_sigaction {
     union {
         void (*_sa_handler)(int);
-        void (*_sa_sigaction)(int, struct siginfo *, void *);
+        void (*_sa_sigaction)(int, siginfo_t *, void *);
     } _u;
     unsigned long sa_flags;
     void (*sa_restorer)(void);
